svglib/svglib.py

```diff
@@ -64,60 +64,188 @@
     'Symbol', 'ZapfDingbats',
 )
 DEFAULT_FONT_NAME = "Helvetica"
-_registered_fonts = {}
+DEFAULT_FONT_WEIGHT = 'normal'
+DEFAULT_FONT_STYLE = 'normal'
 
 logger = logging.getLogger(__name__)
 
 Box = namedtuple('Box', ['x', 'y', 'width', 'height'])
 
 split_whitespace = re.compile(r'[^ \t\r\n\f]+').findall
 
-
-def register_font(font_name, font_path):
+class FontMap:
     """
-    Register a font by name or alias and path to font including file extension.
+    managing the mapping of svg font names to reportlab fonts and registering them in reportlab
     """
-    NOT_FOUND = (None, False)
-    if font_name not in STANDARD_FONT_NAMES and font_name not in _registered_fonts:
+    def __init__(self):
+        self.log = logging.getLogger('{}.{}'.format(__name__, self.__class__.__name__))
+        """ 
+        the map has the form:
+        internal_name as key, and every entry : {'svg_family': 'family_name', 'svg_weight': 'font-weight', 'svg_style': 'font-style', 'rlgFont': 'rlgFontName'}
+        """
+        self._map = {} # for faster searching we use internal keys for finding the matching font
+
+        self.register_default_fonts()
+
+    @staticmethod
+    def build_internal_name(family, weight='normal', style='normal'):
+        """
+        If the weight or style is given, append the capitalized weight and style to the font name.
+        E.g. family="Arial", weight="bold" and style="italic" then the internal name would be "Arial-BoldItalic", this mimics the default fonts naming shema
+        """
+        result_name = family
+        if weight != 'normal' or style != 'normal':
+            result_name += '-'
+        if weight != 'normal':
+            if type(weight) == int:
+                result_name += f'{weight}'
+            else:
+                result_name += weight.lower().capitalize()
+        if style != 'normal':
+            result_name += style.lower().capitalize()
+        return result_name
+
+    @staticmethod
+    def guess_font_filename(basename, weight='normal', style='normal', extension='ttf'):
+        """
+        Try to guess the actual font filename depening on family, weight an style, 
+        this works at least for windows on the "default" fonts like, Arial, courier, Times New Roman etc.
+        """
+        prefix = ''
+        if weight.lower() == 'bold' and style.lower() != 'italic':
```

Comment:
Maybe shorten code by assigning `weight.lower()` and `style.lower()` to temporary vars, and use those.